internal/reference/reference.go

```diff
@@ -33,6 +33,10 @@ type Reference interface {
 	// Delete deletes the value of the reference. If the reference is not
 	// deletable, an error is returned.
 	Delete() error
+	// Rename renames the referenced field, the old field values stay the same
+	// under the new name. Make sure to create a new reference resolver for the
+	// new field name to have access to it.
+	Rename(string) error
```

Comment:
Is the argument supposed just the new name or the full reference? Reading the tests it looks like it's the full reference. Would it be simpler if we just used the name of the field, and not the full reference? E.g. if you have a resolver for `.Payload.After.foo` and call `Rename(bar)`, you'd get `.Payload.After.bar`.

We should also have a way to prevent name clashes. E.g. you might have `.Payload.After.foo` and `.Payload.After.bar` and then you rename `foo` to `bar`. Or you try renaming `.Payload.After.foo` to `.Payload.After`.

Reply:
no it's not the full reference, in the test I'm providing the reference to make it easy to create the reference resolver again and assert the value under the new name.
this method only needs the new name, ex: `Rename(bar)`

Reply:
Ah, I see, I didn't pay much attention to this line: `	err = ref.Rename("newName")`. My mistake!

Reply:
about name clashes:
> Or you try renaming .Payload.After.foo to .Payload.After.

this won't happen because the name provided will replace the name `foo`, so the renaming will rename `.Payload.After.foo` to `.Payload.After.providedName`

> you might have .Payload.After.foo and .Payload.After.bar and then you rename foo to bar

this will result in copying the value from `.Payload.After.foo` to `.Payload.After.bar`, and will delete the `.Payload.After.foo` field, do you mean we should fail in this case?

Reply:
For the last case, I think so. Otherwise, the value of the `bar` field would be lost.

Reply:
done

Reply:
We don't have a good way of accessing the new field. This is also apparent in the test, where we have to provide the new reference string explicitly, that should not be necessary.

I would expect that `Rename` either changes the reference so that it points to the renamed field, or returns another reference that lets me access the renamed field.

So either:
```go
ref, _ := resolver.Resolve(&rec)
err = ref.Rename("newName")
// check error ...
ref.Get() // returns the value under the renamed reference ("newName")
```

or:
```go
ref, _ := resolver.Resolve(&rec)
renamedRef, err := ref.Rename("newName")
// check error ...
ref.Get() // returns an empty value, since the value was deleted
renamedRef.Get() // returns the value under the renamed reference ("newName")
```

I'm leaning towards the second option, as it gives us a bit more flexibility.

Reply:
I went with the second approach, `ref.Get()` will return `nil`, and new reference is for the renamed value.